Client/src/bkr/future/group/modify.py

```diff
@@ -0,0 +1,176 @@
+
+.. option:: --description
+
+   New description of the group.
+
+.. option:: --add-member
+
+   Add a user to the group. This option can be specified multiple
+   times to add more than one user to the group. Should a specified
+   user fail to be added, all subsequent users are ignored.
+
+.. option:: --remove-member
+
+   Remove a user from the group. This option can be specified multiple
+   times to remove more than one user from the group. Should a specified
+   user fail to be removed, all subsequent users are ignored.
+
+.. option:: --grant-owner
+
+   Grant group owner permissions to an existing group member.
+
+.. option:: --revoke-owner
+
+   Remove group owner permissions from an existing group owner.
+
+.. option:: --root-password
+
+   Root password for group jobs.
+
+Common :program:`bkr` options are described in the :ref:`Options
+<common-options>` section of :manpage:`bkr(1)`.
+
+Exit status
+-----------
+
+Non-zero on error, otherwise zero.
+
+Examples
+--------
+
+Modify an existing group 'mygroup' with the new display name 'A new group'::
+
+    bkr group modify --display-name="A new group" mygroup
+
+Modify an existing group 'mygroup' with the new display name 'A new group'
+and new group name 'mynewgroup'::
+
+    bkr group modify --display-name="A new group" --group-name="mynewgroup" mygroup
+
+Add a user with username 'user1' to the group 'mygroup'::
+
+    bkr group modify --add-member user1 mygroup
+
+Remove an existing group member with username 'user1' from the group 'mygroup'::
+
+    bkr group modify --remove-member user1 mygroup
+
+Add an existing group member with username 'user1' as an owner of group 'mygroup'::
+
+    bkr group modify --grant-owner user1 mygroup
+
+Revoke group owner rights from an existing group owner of group 'mygroup' with username 'user1'::
+
+    bkr group modify --revoke-owner user1 mygroup
+
+See also
+--------
+
+:manpage:`bkr(1)`
+
+"""
+from typing import Optional
+from urllib import parse
+
+import click
+from click import ClickException
+
+from bkr.future.api import pass_api
+
+
+@click.command("modify")
+@click.argument("name")
+@click.option("--group-name", help="New name of the group")
+@click.option("--display-name", help="New display name of the group")
+@click.option("--description", help="New description of the group")
+@click.option("--add-member", help="Username of the member to be added", multiple=True)
+@click.option(
+    "--remove-member", help="Username of the member to be removed", multiple=True
+)
+@click.option(
+    "--grant-owner", help="Username of the member to grant owner rights", multiple=True
+)
+@click.option(
+    "--revoke-owner",
+    help="Username of the member to revoke owner rights",
+    multiple=True,
+)
+@click.option("--root-password", help="Root password used for group jobs")
+@pass_api
+def modify(
+    api,
+    name: str,
+    group_name: Optional[str],
+    display_name: Optional[str],
+    description: Optional[str],
+    add_member: tuple,
```

Comment:
should not add/remove/ownership be part of the member command, like, changing it to `bkr group members list`, and then having `bkr group members add`?

Reply:
This is definitely a good idea. We can shuffle the files when CLI is done, before the next release. But I believe, for now, it is sufficient. We can see that functionality is there and working as expected.

Reply:
oki, sounds like a plan :D